src/pages/Preferences.vue

```diff
@@ -58,6 +61,8 @@ export default Vue.extend({
   methods:  {
     navChanged(tabName: string) {
       this.currentNavItem = tabName;
+      ipcRenderer.invoke('settings-write',
+        { preferences: { currentNavItem: this.currentNavItem } } );
```

Comment:
You might want to consider dispatching the `commitPreferences` action with a payload. You should be able to do something similar to what we do for muting diagnostics to get the same result:

https://github.com/rancher-sandbox/rancher-desktop/blob/d69aa4567c1cae71bc604bf678d781991d8762a2/src/store/diagnostics.ts#L121-L128

Reply:
Hi @rak-phillip , thanks for replying. I have changed the code as you suggested, please check my new [commit](https://github.com/rancher-sandbox/rancher-desktop/pull/3010/commits/87cb9b919b44ce2af1371f275271aa17160df7de) , I will squash it in the case it is ok.

I have a concern. 
Fetching `settings` through https request means the variable `currentNavItem` is updated in async way, so it is initialized with default value `Application` and quite immediately it receive the last saved value from the async request.

https://github.com/rancher-sandbox/rancher-desktop/blob/87cb9b919b44ce2af1371f275271aa17160df7de/src/pages/Preferences.vue#L23-L27

As result, the preferences dialog is initiated with `Application` page and then immediately switch to the page matching with current `settings`. Even though it's not visible to human eye, I don't know if it could be a problem...

Reply:
Sorry, I missed this question and didn't notice it again until now.

We should be able to get around this issue if we refactor similar to what's described in https://github.com/rancher-sandbox/rancher-desktop/pull/3010#discussion_r1001060407

However, I did notice while testing that there appears to be a bug in the tabbed component that looks to be rebuilding the entire DOM on first render. I've created #3232 so we can address